src/toolkit/gis.h

```diff
@@ -0,0 +1,158 @@
+/// @class GIS
+/// The GIS class is a basic class that stores the geographic location of each
+/// agent in latitude and longitude and follows the ISO
+/// 6709 standard. Allows string expression of GIS objects as specified in
+/// ANNEX H of ISO 6709 found here: 'https://www.iso.org/standard/39242.html'
+/// or a quick simplified version found here
+/// 'https://en.wikipedia.org/wiki/ISO_6709'
+///
+/// Brief Description of ISO 6709:
+/// 1. Latitude comes before Longitude
+/// 2. North latitude is positive
+/// 3. East longitude is positive
+/// 4. Fraction of degrees (decimal values) is prefered in digital data
+/// exchange.
+///
+/// The class is adapted from 'https://github.com/jaime-olivares/coordinate'
+/// under the MIT License.
+///
+/// Longitude and Latitude is stored as seconds of degrees. This was also
+/// adapted from 'https://github.com/jaime-olivares/coordinate'. This, according
+/// to the jaime-olivares, allows the coordinate elements such as degrees,
+/// minutes, and seconds to "remain on the integral portion of values, with the
+/// exception of decimal of seconds, avoiding loss of precision." This is
+/// calculated by multiplying decimal degrees by 3600.
+/// example: 05.2169 -> 18780.84
+
+class GIS {
+ public:
+  /// The default constructor for GIS. This will creat an object with
+  /// lat =0, and long=0.
+  GIS();
+
+  /// GIS constructor with latitude and longditude as degrees as a decimal
+  /// number.
+  /// @param deciaml_lat latitude expressed in degrees as a decimal number.
+  /// @param decimal_lon longditude expressed in degrees as a decimal number.
+  GIS(float decimal_lat, float decimal_lon);
+
+  /// The default destructor for GIS
+  ~GIS();
+
+  /// Returns the current latitude.
+  /// @return decimal representation of latitude.
+  float get_latitude() const;
+
+  /// Returns the current longitude.
+  /// @return decimal representation of longitude.
+  float get_longitude() const;
+
+  /// Sets a new latitude
+  /// @param lat latitude expressed in decimal degrees.
+  void set_latitude(float lat);
+
+  /// Sets a new longitude
+  /// @param lon longitude expressed in decimal degrees.
+  void set_longitude(float lon);
+
+  /// Returns the distance (in km) between this GIS object and the target GIS
+  /// object.
+  /// @param target the GIS object some distnace away from the current one
+  /// @return distance between this and target in kilometers
+  double Distance(GIS target) const;
+
+  /// Converts GIS location into a string expression that follows ISO 6709 Annex
+  /// H.
+  /// @param return_format
+  /// @parblock
+  ///       The format of output. Options are '1', '2', or '3':
+  ///
+  ///     1:   return in degrees format
+  ///     2:   return in degrees and minutes format
+  ///     3:   return in egrees minutes seconds format
+  /// @endparblock
+  /// @return String representation of the GIS object that complies with ISO6709
+  /// Annex H
+  std::string ToString(int return_format) const;
```

Comment:
I recommend having an enum that represents these options, so the user code is more readable. Also, I recommend having a default value (probably of 1) so that users don't have to worry about always supplying this value if they just want to get a string.